prod/init-services.sh

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env bash
+
+set -eu
+
+trapdoor(){
+  # Define a list of .env files to check in service directories
+  ENV_FILES=(
+    "./services/wiki/php.env"
+    "./services/wiki/mariadb.env"
+    "./services/online/php.env"
+    "./services/online/mariadb.env"
+    "./services/click/php.env"
+    "./services/click/mariadb.env"
+    "./services/danger/php.env"
+    "./services/danger/mariadb.env"
+  )
+
+  # Check if any of the .env files already exist
+  for env_file in "${ENV_FILES[@]}"; do
+    if [ -f "$env_file" ]; then
+      echo "Environment file $env_file already exists. Exiting."
+      exit 1
+    fi
+  done
+
+  # If no .env files are found, proceed with the script
+  echo "No .env files found. Proceeding with the script."
+}
+
+if [[ $(docker network ls | grep "$1") ]]; then
+  echo "Traefik backend already exists"
+else
+  docker network create traefik-backend
+fi
+
+# Generate a random SHA-512 hash
+generate_random_pass() {
+  pwgen -s 32 1
+}
+
+count_dir() {
+    SERVICE_ITEMS=($(ls ./services/))
+}
+
+# Create env var for services as needed
+export_secrets() {
+  export ENV_TAG="prod"
+
+  # Iterate through all service items
+  for item in "${SERVICE_ITEMS[@]}"; do
+    # Replace hyphens with underscores in the service name
+    local service_name="${item//-/_}"
+
+    # Generate random passwords and declare service-specific variables
+    declare -g "${service_name^^}_MARIADB_ROOT_PASSWORD=$(generate_random_pass)"
+    declare -g "${service_name^^}_MARIADB_PASSWORD=$(generate_random_pass)"
+  done
+
+  # Dynamically set the shared global variables for all services
+  for item in "${SERVICE_ITEMS[@]}"; do
+    # Replace hyphens with underscores in the service name for the global variable name
+    local service_name="${item//-/_}"
+
+    # Export global variables dynamically for each service
+    export "${service_name^^}_MARIADB_ROOT_PASSWORD"
+    export "${service_name^^}_MARIADB_PASSWORD"
+  done
+
+  # Export other secrets as needed
+  export DB_PASSWORD_WIKI=$(generate_random_pass)
```

Comment:
This DB variable is used across all projects, not just the wiki. For clarity this should be updated to `DB_PASSWORD` to avoid confusion in the future